### PR TITLE
[RFC] Fix Issue-1016 Armdev - build and run x86_64 without issues.

### PR DESCRIPTION
**Pull request by chl337**
Addresses [issue-1016](https://github.com/QuarkContainer/Quark/issues/1016):

-  armdev should not cause issues on x86_64.

Result of local test:

> docker run  --runtime=quark_d -ti --entrypoint /bin/bash ubuntu
> root@302a992d1d24:/# whoami
> root

### CODE REVIEW

File: qlib/kernel/vcpu.rs [hunk 1]
```diff
@@ -73,16 +73,14 @@ pub fn RegisterSysCall(addr: u64) {
...
+#[cfg(target_arch = "x86_64")]
 pub fn SetTLS(addr: u64) {
     //println!("SetFs from {:x} to {:x}", GetFs(), addr);
     WriteMsr(MSR::MSR_FS_BASE as u32, addr);
     //println!("the input value is {:x}, the get fs result is {:x}", addr, ReadMsr(MSR::MSR_FS_BASE as u32));
 }
 
 #[cfg(target_arch = "aarch64")]
-#[inline]
 pub fn SetTLS(addr: u64) {
```
[the remark this answers is not in the document]
Comment (chl337) [reply qlib/kernel/vcpu.rs:84]: They are just hints for the compiler and for such one-liners, they are not really necessary. I think we should use these attributes only when deemed necessary.

File: qvisor/src/runc/runtime/vm.rs [hunk 2]
```diff
@@ -333,31 +333,23 @@ impl VirtualMachine {
                 MemoryDef::PHY_LOWER_ADDR + 64 * MemoryDef::ONE_MB + 2 * MemoryDef::ONE_GB;
             vms.pageTables = PageTables::New(&vms.allocator)?;
 
-            let mut opts = addr::PageOpts::Zero();
-            opts.SetPresent().SetWrite().SetGlobal();
-            #[cfg(target_arch = "aarch64")]
-            opts.SetMtNormal().SetDirty().SetAccessed();
-
+            let opts: PageOpts = PageEntryType::page_options(PageEntryType::Kernel).unwrap();
```
Comment (CharlyYu) [inline qvisor/src/runc/runtime/vm.rs:336]: Maybe `page_options` don't need to return an Option, it is ok to return a default PageOpts, add unwrap() here will add the chance to panic.

Comment (chl337) [reply qvisor/src/runc/runtime/vm.rs:336]: Is thought of as a precaution, in the case of non-implemented features. Normally it should be checked if _something_ is returned and decided accordingly.
Of course, I can change it if it is deemed _"bad practice"_.

**Comment by CharlyYu** [comment]
It does not compile in aarch64.

**Comment by chl337** [comment]
@CharlyYu could you please post a log with the error?

**Comment by shrik3** [comment]
@CharlyYu @chl337 

> It does not compile in aarch64.

can confirm, seems like a wrong import:

build qvisor debug
```
error[E0432]: unresolved import `super::utils`                                       
  --> src/qlib/kernel/arch/./aarch64/context.rs:23:12                                
   |                                                                                 
23 | use super::utils::MMapRand;                                                     
   |            ^^^^^ could not find `utils` in `super`                              
                                                                                     
error[E0433]: failed to resolve: use of undeclared crate or module `addr`            
   --> src/runc/runtime/vm.rs:354:21                                                 
    |                                                                                
354 |                     addr::Addr(MemoryDef::HYPERCALL_MMIO_BASE),                
    |                     ^^^^ use of undeclared crate or module `addr`              
                                                                                     
error[E0433]: failed to resolve: use of undeclared crate or module `addr`            
   --> src/runc/runtime/vm.rs:355:21                                                 
    |                                                                                
355 |                     addr::Addr(MemoryDef::HYPERCALL_MMIO_BASE + 0x1000),       
    |                     ^^^^ use of undeclared crate or module `addr`              
                                                                                     
error[E0433]: failed to resolve: use of undeclared crate or module `addr`            
   --> src/runc/runtime/vm.rs:356:21                                                 
    |                                                                                
356 |                     addr::Addr(MemoryDef::HYPERCALL_MMIO_BASE),                
    |                     ^^^^ use of undeclared crate or module `addr`              
                                                                                     
error[E0425]: cannot find value `opts` in this scope                                 
   --> src/qlib/addr.rs:379:3                                                        
    |                                                                                
379 |         opts.SetWrite().SetGlobal().SetPresent().SetAccessed()                 
    |      
```

build qkernel debug

```
error[E0432]: unresolved import `super::utils`                                       
  --> src/qlib/kernel/arch/./aarch64/context.rs:23:12                                
   |                                                                                 
23 | use super::utils::MMapRand;                                                     
   |            ^^^^^ could not find `utils` in `super`                              
                                                                                     
error[E0425]: cannot find value `opts` in this scope                                 
   --> src/qlib/addr.rs:379:3                                                        
    |                                                                                
379 |         opts.SetWrite().SetGlobal().SetPresent().SetAccessed()                 
    |         ^^^^ not found in this scope
```

**Comment by chl337** [comment]
@CharlyYu, @QuarkContainer commit [6b8cdf2](https://github.com/QuarkContainer/Quark/commit/6b8cdf25d5a23e526baf49b7aab23cdfa568d61d) seems to break x86_64 functionality on branch-armdev (compare with [57d88c1](https://github.com/QuarkContainer/Quark/commit/57d88c1e4518098f3c7cc5a055da47e0ee995870)), [testing like above](https://github.com/QuarkContainer/Quark/pull/1017#issue-1989652631) will cause a GeneralProtectionFault (see generated quark.log).
Steps:

1. For building on x86_64: git apply [build_x86_on_armdev](https://jmp.sh/h7tF4Mtw)
2. make clean
3. make debug
4. make install
5. docker run --runtime=quark_d -ti --entrypoint /bin/bash ubuntu

I would appreciate some help on this.

**Pull request closed without merging**